### PR TITLE
pykeyvi.[cpp/pyx] on the fly generation

### LINKED ISSUE #187: remove generated pykeyvi.[pyx, cpp] files from git and generate them on the fly during build

### PR DESCRIPTION
**Pull request by narekgharibyan**
@hendrikmuhs 

Fixes #187

### CODE REVIEW

File: pykeyvi/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+msgpack-python
+autowrap
```
Comment (hendrikmuhs) [inline pykeyvi/requirements.txt]: I wonder if we should pin autowrap to a version, as this introduces a build time dependency to a API that does not look official to me.

**Comment by hendrikmuhs** [comment]
@narekgharibyan can you update the Readme in the pykeyvi folder accordingly.

**Comment by hendrikmuhs** [comment]
I guess you already saw that sdist is broken. IMO sdist should work the same way as before meaning shipping the cpp file (I think the pyx is not required) and not depending on autowrap.

File: pykeyvi/requirements.txt (hunk 1 above)
Comment (narekgharibyan) [inline pykeyvi/requirements.txt]: Good point.
I think it's better to ask `autowrap` guys to confirm or add an official API.

**Comment by narekgharibyan** [comment]
Yup! will update Readme.
Yeah, it broke because of my fault, already pushed the fix.

**Comment by narekgharibyan** [comment]
@hendrikmuhs 
regarding to including `cpp` file into `sdist` package
I think we need something like:
```
try:
   import autowrap
   ... then generate source ... 
catch: 
   use shipped version
``` 

As I see this is common approach.

**Comment by hendrikmuhs** [comment]
I suggest to follow this guidance:

http://docs.cython.org/en/latest/src/reference/compilation.html#distributing-cython-modules

It recommends to use the cpp file and provide cython(or autowrap) only explicit by a cmdline option.

**Comment by narekgharibyan** [comment]
Agree that for `sdist` it should use generated source file by default (even I guess it should be mandatory).

Can you please have a look [here](https://github.com/narekgharibyan/keyvi/commit/2077facccad2c52f8a297576679b205a588c1239) ? 
This addresses both problems:
1) no generated files are committed to Git, and are updated on the fly when there is a change.
2) `sdist` does not depend on `autowrap` and uses shipped `pykeyvi.cpp`

**Comment by hendrikmuhs** [comment]
@narekgharibyan :+1: This change looks good to me.

**Comment by hendrikmuhs** [comment]
@narekgharibyan What about the autowrap and documentation issue?

**Comment by narekgharibyan** [comment]
Documentation is on the way (my bad) and need to contact `autowrap` guys. 
Thought do not wait till they confirm or ...

**Comment by hendrikmuhs** [comment]
agreed, but till then can you simply pin the version to 0.9.2 ?

**Comment by narekgharibyan** [comment]
Sure!

**Pull request merged**